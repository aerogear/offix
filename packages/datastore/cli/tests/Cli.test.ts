import * as util from 'util';
import { mkdirSync, rmdirSync, readFileSync, existsSync } from "fs";
import { join } from 'path';

beforeEach(() => {
  if (!existsSync("./output/"))
    mkdirSync("./output/")
});
afterEach(() => rmdirSync("./output/", { recursive: true }));

test("generate", async () => {
  const exec = util.promisify(require('child_process').exec);
  const cliExec = join(__dirname, '../bin/offix.js');
  const schemaPath = join(__dirname, './mock.graphql');

  await exec(`node ${cliExec} generate --schema ${schemaPath} --outputPath output/`);
  const schema = readFileSync("./output/schema.json").toString();
  const index = readFileSync("./output/index.ts").toString();
  const types = readFileSync("./output/types.ts").toString();

  expect(types).toMatchSnapshot();
  expect(schema).toMatchSnapshot();
  expect(index).toMatchSnapshot();
});
