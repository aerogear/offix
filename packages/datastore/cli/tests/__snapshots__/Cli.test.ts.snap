// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate with defaults 1`] = `
"export interface Note {
    _id: string;
    title: string;
    description: string;
    comments: string[]
    _version: number;
}

export type NoteCreate = Omit<Note, \\"_id\\">;
export type NoteChange =  Pick<Note, \\"_id\\"> & Partial<NoteCreate>;

export interface Comment {
    _id: string;
    title: string;
    description: string;
    noteComment: string
    _version: number;
}

export type CommentCreate = Omit<Comment, \\"_id\\">;
export type CommentChange =  Pick<Comment, \\"_id\\"> & Partial<CommentCreate>;
"
`;

exports[`generate with defaults 2`] = `
"{
  \\"Note\\": {
    \\"name\\": \\"Note\\",
    \\"version\\": 1,
    \\"type\\": \\"object\\",
    \\"primaryKey\\": \\"_id\\",
    \\"properties\\": {
      \\"_id\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_id\\",
        \\"isRequired\\": true,
        \\"primary\\": true
      },
      \\"title\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"title\\",
        \\"isRequired\\": true
      },
      \\"description\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"description\\",
        \\"isRequired\\": true
      },
      \\"_version\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_version\\",
        \\"isRequired\\": true
      }
    }
  },
  \\"Comment\\": {
    \\"name\\": \\"Comment\\",
    \\"version\\": 1,
    \\"type\\": \\"object\\",
    \\"primaryKey\\": \\"_id\\",
    \\"properties\\": {
      \\"_id\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_id\\",
        \\"isRequired\\": true,
        \\"primary\\": true
      },
      \\"title\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"title\\",
        \\"isRequired\\": true
      },
      \\"description\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"description\\",
        \\"isRequired\\": true
      },
      \\"_version\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_version\\",
        \\"isRequired\\": true
      }
    }
  }
}"
`;

exports[`generate with defaults 3`] = `
"import { GeneratedModelSchema } from \\"offix-datastore\\";
import jsonSchema from \\"./schema.json\\";

export const schema = jsonSchema as GeneratedModelSchema;

export * from \\"./types\\";
"
`;

exports[`generate with positional arguments 1`] = `
"export interface Note {
    _id: string;
    title: string;
    description: string;
    comments: string[]
    _version: number;
}

export type NoteCreate = Omit<Note, \\"_id\\">;
export type NoteChange =  Pick<Note, \\"_id\\"> & Partial<NoteCreate>;

export interface Comment {
    _id: string;
    title: string;
    description: string;
    noteComment: string
    _version: number;
}

export type CommentCreate = Omit<Comment, \\"_id\\">;
export type CommentChange =  Pick<Comment, \\"_id\\"> & Partial<CommentCreate>;
"
`;

exports[`generate with positional arguments 2`] = `
"{
  \\"Note\\": {
    \\"name\\": \\"Note\\",
    \\"version\\": 1,
    \\"type\\": \\"object\\",
    \\"primaryKey\\": \\"_id\\",
    \\"properties\\": {
      \\"_id\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_id\\",
        \\"isRequired\\": true,
        \\"primary\\": true
      },
      \\"title\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"title\\",
        \\"isRequired\\": true
      },
      \\"description\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"description\\",
        \\"isRequired\\": true
      },
      \\"_version\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_version\\",
        \\"isRequired\\": true
      }
    }
  },
  \\"Comment\\": {
    \\"name\\": \\"Comment\\",
    \\"version\\": 1,
    \\"type\\": \\"object\\",
    \\"primaryKey\\": \\"_id\\",
    \\"properties\\": {
      \\"_id\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_id\\",
        \\"isRequired\\": true,
        \\"primary\\": true
      },
      \\"title\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"title\\",
        \\"isRequired\\": true
      },
      \\"description\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"description\\",
        \\"isRequired\\": true
      },
      \\"_version\\": {
        \\"type\\": \\"string\\",
        \\"key\\": \\"_version\\",
        \\"isRequired\\": true
      }
    }
  }
}"
`;

exports[`generate with positional arguments 3`] = `
"import { GeneratedModelSchema } from \\"offix-datastore\\";
import jsonSchema from \\"./schema.json\\";

export const schema = jsonSchema as GeneratedModelSchema;

export * from \\"./types\\";
"
`;
